File: js-pr/56/javascript/code-quality-readability-maintainability/calculator_27.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function calculateComplexFormula(a, b, c, d, e, f, g, h, i, j) {
    const allValuesPresent = [a, b, c, d, e, f, g, h, i, j].every(val => val !== undefined && val !== null);
    
    if (!allValuesPresent) {
        return 0;
    }
    
    const term1 = a + b * c;
    const term2 = d / e;
    const term3 = f * g;
    const term4 = h + i * j;
    
    return term1 - term2 + term3 - term4;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/calculator_27.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def calculate_complex_formula(a, b, c, d, e, f, g, h, i, j):
    if not all([a, b, c, d, e, f, g, h, i, j]):
        return 0
    
    term1 = a + b * c
    term2 = d / e if e != 0 else 0
    term3 = f * g
    term4 = h + i * j
    
    return term1 - term2 + term3 - term4
# {/fact}